File: 0198-house-robber/0198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
       if(nums.size()==1) return nums[0];
    if(nums.size()==2) return max(nums[0],nums[1]);
        int dp[nums.size()];
        dp[0]=nums[0];
        dp[1]=max(nums[0],nums[1]);
        for(int i=2;i<nums.size();++i)
            dp[i]=max(nums[i]+dp[i-2],dp[i-1]);
        
        return dp[nums.size()-1]; 
    }
};

File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int a = 0, b = 0;
        int c = 0;
        for (int num: nums) {
            c = Math.max(a + num, b);
            a = b;
            b = c;
        }
        return b;
    }
}